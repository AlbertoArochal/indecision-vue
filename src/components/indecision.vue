<template>
    <div>
        <img :src="image" alt="bg" />
        <div class="bg-dark">
            <h1>Indecision</h1>
            <div class="indecision__container">
                <input
                    type="text"
                    placeholder="Hazme una pregunta"
                    v-model="question"
                />
                <p>Recuerda terminar con un signo de interrogaci√≥n</p>
            </div>
            <h2>{{ question }}</h2>
            <div v-if="isValidQuestion">
                <p>{{ answer }}</p>
            </div>
        </div>
    </div>
</template>
<script lang="ts">
export default {
    props: {},
    name: "Indecision-vue",
    data() {
        return {
            question: "",
            answer: "",
            image: "",
            isValidQuestion: false,
        };
    },
    watch: {
        question(value: string) {
            this.isValidQuestion = false;
            if (!value.includes("?")) return;
            this.isValidQuestion = true;
            this.getAnswer();
        },
    },
    methods: {
        async getAnswer() {
            this.answer = "Loading...";
            const { answer, image } = await fetch("https://yesno.wtf/api").then(
                (response) => response.json()
            );

            this.answer = answer === "yes" ? "Si" : "No!";
            this.image = image;
        },
    },
};
</script>
