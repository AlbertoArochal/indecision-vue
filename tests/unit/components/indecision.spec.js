import indecision from "@/components/indecision";
import { shallowMount } from "@vue/test-utils";

describe("indecision.vue", () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallowMount(indecision);
    });
    it("should match snapshot", () => {
        const h2 = wrapper.find("h2");
        expect(h2.text()).toBe("Indecision");
    });
});
