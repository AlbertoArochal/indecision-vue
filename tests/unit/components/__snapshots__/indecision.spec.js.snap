// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indecision.vue should match snapshot 1`] = `[Function]`;

exports[`indecision.vue should match snapshot 1`] = `
<div><img src="" alt="bg">
  <div class="bg-dark">
    <h1>Indecision</h1>
    <div class="indecision__container"><input type="text" placeholder="Hazme una pregunta">
      <p>Recuerda terminar con un signo de interrogaci√≥n</p>
    </div>
    <h2></h2>
    <!--v-if-->
  </div>
</div>
`;
